internal/blueprint/customizations_test.go

```diff
@@ -0,0 +1,158 @@
+package blueprint
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestCustomizationsHostname(t *testing.T) {
+
+	var myHostname = "MyHostname"
+
+	TestCustomizations := Customizations{
+		Hostname: &myHostname,
+	}
+
+	bpOrig := Blueprint{
+		Name:           "customizations-test",
+		Description:    "Testing customizations functions",
+		Version:        "0.0.1",
+		Customizations: &TestCustomizations,
+	}
+
+	assert.Equalf(t, &myHostname, TestCustomizations.GetHostname(), "The hostname is different")
+	assert.Equalf(t, &myHostname, bpOrig.Customizations.GetHostname(), "The hostname is different")
+
+}
+
+func TestKernelCustomization(t *testing.T) {
+
+	myKernel := KernelCustomization{
+		Append: "--test",
+	}
+
+	TestCustomizations := Customizations{
+		Kernel: &myKernel,
+	}
+
+	assert.Equalf(t, &myKernel, TestCustomizations.GetKernel(), "The kernel is different")
+}
+
+func TestSSHKeyCustomization(t *testing.T) {
+
+	mySSHKeys := []SSHKeyCustomization{
+		SSHKeyCustomization{
+			User: "test-user",
+			Key:  "test-key",
+		},
+	}
+	TestCustomizations := Customizations{
+		SSHKey: mySSHKeys,
+	}
+
+	assert.Equal(t, mySSHKeys[0].User, TestCustomizations.GetUsers()[0].Name, "The user names are different")
+	assert.Equal(t, mySSHKeys[0].Key, *TestCustomizations.GetUsers()[0].Key, "The users keys are different")
```

Comment:
IMO assigning the actual value to a temporary variable and asserting on that will be more clear. However I don't understand the following:
- why does the 2nd call to GetUsers uses `*` in front of the result? I'm sure the answer has to do with pointers but from what I can tell both Name and Key fields have the same data type and I don't understand why that applies to one of them and not the other.

Reply:
Well it only works when I use a pointer to that and that seems to be because of how `GetUsers()` function is written.. It actually returns `Users struct` where it appends the `SSHKeys` if they exist. `Name` is just a regular `string` but `Key` is a `*string` so that's probably why.

Reply:
I see the difference now. Thanks.